src/OpenTelemetry.Sampler.AWS/AWSXRayRemoteSampler.cs

```diff
@@ -0,0 +1,84 @@
+// <copyright file="AWSXRayRemoteSampler.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Threading;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Sampler.AWS;
+
+/// <summary>
+/// Remote sampler that gets sampling configuration from AWS X-Ray.
+/// </summary>
+public sealed class AWSXRayRemoteSampler : Trace.Sampler, IDisposable
+{
+    internal AWSXRayRemoteSampler(TimeSpan pollingInterval, string endpoint)
+    {
+        this.PollingInterval = pollingInterval;
+        this.Endpoint = endpoint;
+        this.Client = new AWSXRaySamplerClient(endpoint);
+
+        // execute the first update right away
+        this.RulePollerTimer = new Timer(this.GetAndUpdateSampler, null, TimeSpan.Zero, this.PollingInterval);
+    }
+
+    internal TimeSpan PollingInterval { get; }
+
+    internal string Endpoint { get; }
+
+    internal AWSXRaySamplerClient Client { get; }
+
+    internal Timer RulePollerTimer { get; }
+
+    /// <summary>
+    /// Initializes a <see cref="AWSXRayRemoteSamplerBuilder"/> for the sampler.
+    /// </summary>
+    /// <returns>an instance of <see cref="AWSXRayRemoteSamplerBuilder"/>.</returns>
+    public static AWSXRayRemoteSamplerBuilder Builder()
+    {
+        return new AWSXRayRemoteSamplerBuilder();
+    }
+
+    /// <inheritdoc/>
+    public override SamplingResult ShouldSample(in SamplingParameters samplingParameters)
+    {
+        // TODO: add the actual functionality for sampling.
+        throw new System.NotImplementedException();
+    }
+
+    /// <inheritdoc/>
+    public void Dispose()
+    {
+        this.Dispose(true);
+        GC.SuppressFinalize(this);
+    }
+
+    private void Dispose(bool disposing)
+    {
+        if (disposing)
+        {
+            this.RulePollerTimer?.Dispose();
+            this.Client?.Dispose();
+        }
+    }
+
+    private async void GetAndUpdateSampler(object? state)
```

Comment:
You should probably be using async Task instead of async void here.

Reply:
The `GetAndUpdateSampler` method here is a [`TimerCallback`](https://learn.microsoft.com/en-us/dotnet/api/system.threading.timercallback?redirectedfrom=MSDN&view=net-6.0) delegate which needs to have a `void` return type.

I believe that although generally it is preferred to use async Task instead of async void, but with even handlers like this, it is okay to use the latter.